src/main/java/org/testng/TestNG.java

```diff
@@ -1365,7 +1375,8 @@ protected void configure(CommandLineArgs cla) {
     setOutputDirectory(cla.outputDirectory);
 
     if (cla.testNames != null) {
-      setTestNames(Arrays.asList(cla.testNames.split(",")));
+      //Ensure that dynamic list operations supported
+      setTestNames(new ArrayList<>(Arrays.asList(cla.testNames.split(","))));
```

Comment:
Do we need a `new ArrayList<>` creation here especially when `Arrays.asList()` itself creates a `List` ?

Reply:
Yes, we need. Array.asList() wraps an array in the list interface. The list is still backed by the array. Arrays are a fixed size - they don't support adding or removing elements, so the wrapper can't either.

Reply:
@eyehwan - Yes I understand that. But I was more trying to understand how and where is this list being altered. And if its being altered, why so. `testnames` once split by `,` should have only been used and this list serves as a `list representation of the split string`. So if are trying to add data to it, then something is going wrong somewhere.

Reply:
XmlSuiteUtils::getMissMatchedTestNames was implemented by me would use remove function.

Reply:
@eyehwan - We should be able to easily get past it, by having the `getMissMatchedTestNames` work with a clone of the `testNames` list, which it prunes and returns back, and gets used later. If we need to the `cla` value of `testNames` to be updated, then we should be invoking `setTestNames()` and updating that list, rather than alter the list.

Reply:
>So if are trying to add data to it, then something is going wrong somewhere.
You remind me, thanks @krmahadevan. I will provide another solution, so see if it works.

Reply:
@krmahadevan I change the way of handling list iterator on ` XmlSuiteUtils::getMissMatchedTestNames` by `addAll`, which avoids impacting on m_testNames.

Reply:
You should be able to remove the `new ArrayList` now, right?

Reply:
Okay